packages/vitest/src/types/coverage.ts

```diff
@@ -157,40 +157,62 @@ export interface BaseCoverageOptions {
   skipFull?: boolean
 
   /**
-   * Check thresholds per file.
-   * See `lines`, `functions`, `branches` and `statements` for the actual thresholds.
+   * Threshold for statements, lines, branches and functions
    *
-   * @default false
-   */
-  perFile?: boolean
-
-  /**
-   * Threshold for lines
+   * Default value is `undefined` for each property.
    *
-   * @default undefined
-   */
-  lines?: number
-
-  /**
-   * Threshold for functions
+   * @example
    *
-   * @default undefined
-   */
-  functions?: number
-
-  /**
-   * Threshold for branches
+   * ```ts
+   * {
+   *   // Thresholds for all files
+   *   functions: 95,
+   *   branches: 70,
    *
-   * @default undefined
+   *   // Thresholds for utilities
+   *   'src/utils/**.ts': {
+   *     lines: 100,
+   *     statements: 95,
+   *   }
+   * }
+   * ```
    */
-  branches?: number
-
-  /**
-   * Threshold for statements
-   *
-   * @default undefined
-   */
-  statements?: number
+  thresholds?: {
+    /** Set global thresholds to `100` */
+    100?: boolean
+
+    /** Check thresholds per file. */
+    perFile?: boolean
+
+    /**
+     * Update threshold values automatically when current coverage is higher than earlier thresholds
+     *
+     * @default false
+     */
+    autoUpdate?: boolean
+
+    statements?: number
+    functions?: number
+    branches?: number
+    lines?: number
+
+    /**
+     * Thresholds for files matching the glob pattern
+     *
+     * @example
+     *
+     * ```ts
+     * {
+     *  'src/utils/**.ts': {
+     *    statements: 80,
+     *    functions: 95,
+     *    branches: 100,
+     *    lines: 50,
+     * }
+     * ```
+     */
+    [glob: string]: unknown
```

Comment:
These can also be typed: https://www.typescriptlang.org/play?#code/C4TwDgpgBAKgFgJwgZzgewDYBNkCECGy0AvFAN4BQUUyw+wEAthAHbDID8AXFCwK6MARhARUoAMz4sAxsACWaFpx78hIsYIT4ZcFN14Dho6hjks9Kw+uoBGAAx39gtJgjaxkBADE5GCE5c-dwBfClBIWEQUdGxkKFJ4JFRMHAIiKAAfKAAKMigAbQBzDDRBHloEM0KAXR5E6JS8QmhgqAAySKSY1OaASgoB6UVaKCGANxF8QogePOAo5Ni6he641tJKanmuxtmxagB6A6gAcRLBfAwaCGB5FkLkfZo6BmY2ZB57OwAaJ8kZeTDT4OX7UaiabTSXQfKBfUFg0zmGFwp5fHjABB8CDw6ieHx+dGY7EDMFQI6nc5QC5ELDXW5VR6kgBEACoWQc+PIMMgDmymXtSdRaPQmKx2MCfk9qP9ZAolBKcWCIToUAqpVBEarYSD1WioBisYrcSJ8TMJJciIrgr8nqz2WApiheSz+eR1ZqYQBWSWCqB43xm8QW4mk61iMOhIA

Reply:
What is stopping us from using these types?

Reply:
I'm planning to rewrite the current implementation to use the typings shown in the linked TS playground. Nothing is stopping from using those. 

Earlier I thought using such typed dynamic keys in `interface` would not be possible in Typescript, but someone in Vitest Discrod provided solution that works really nice. Intellisense and type checking works even for those `[glob: string]` types there. I received these types just this morning and haven't had the time to implement it yet. Droppped the link here for later as notification.